WEB-4108 niapi promo numbers does not support createupdate by idfromn…

collin-mcgrath opened the pull request:
https://developers.invoca.net/en/1810-web-4108_niapi_promo_numbers_does_not_support_createupdate_by_idfromnetwork/

collin-mcgrath committed:
WEB-4108 niapi promo numbers does not support createupdate by idfromnetwork: Add documentation for new promo_numbers_by_id endpoint

collin-mcgrath commented:
@nburwell Let me know what you think of these changes and I can update Adv, Aff, AdvCam after that initial round of CR.

nburwell commented:
@collin-mcgrath I would love to review this by viewing a private version on our dev portal .. much better than parsing the markup here. If that is not possible by EOD (due to RTD issues) I will have to pass CR of this to someone else if we want it before Monday

collin-mcgrath committed:
WEB-4108 niapi promo numbers does not support createupdate by idfromnetwork: Use correct promo_number identifier

collin-mcgrath committed:
WEB-4108 niapi promo numbers does not support createupdate by idfromnetwork: Remove redundant documentation

collin-mcgrath commented:
~~The dev portal has the private version now, but it doesn't seem to be viewable. This may require downloading and building locally for review.~~

https://developers.invoca.net/en/1810-web-4108_niapi_promo_numbers_does_not_support_createupdate_by_idfromnetwork/

collin-mcgrath committed:
WEB-4108 niapi promo numbers does not support createupdate by idfromnetwork: Describe Read the Docs build workaround for new branches

- Should be remove when https://github.com/rtfd/readthedocs.org/issues/4450 is completed.

rjkearns commented on source/api_documentation/network_integration/promo_numbers/_post_promo_numbers_by_id.rst:
```diff
@@ -0,0 +1,210 @@
+
+    {
+      "id": 12,
+      "id_from_network": "12",
```
This should be 1234, right?

collin-mcgrath replied:
Good catch.

rjkearns commented on source/api_documentation/network_integration/promo_numbers/_post_promo_numbers_by_id.rst:
```diff
@@ -0,0 +1,210 @@
+      "custom_data": {
+        "channel": "Online lead"
+      }
+    }
```
Are all these examples needed? Is it possible to just show one example using the `id_from_network` param?

collin-mcgrath replied:
I just copied it verbatim from the one below. I'm inclined to leave it since this section is now above the other one (per the ticket).

collin-mcgrath replied:
I agree that it might be a little excessive, but I don't really want to remove something from a live version.

rjkearns replied:
From a live version? I was referring to just including one example in the new `id_from_network` create panel example. I'm fine the way it is though.

rjkearns reviewed (commented):
Left a couple of questions. 

I did notice an issue, which might be a ReadTheDocs issue? I was unable to open the `GET /promo_numbers_by_id/<id_from_network>` panel. ReadTheDocs kept opening the `GET /promo_numbers_by_id/` panel instead. This only happened for the `promo_numbers_by_id` section

joroeinvoca commented on source/api_documentation/network_integration/promo_numbers/_put_promo_number_by_id_with_id.rst:
```diff
@@ -0,0 +1,55 @@
+
+  Endpoint:
+
+  ``https://invoca.net/api/@@NETWORK_API_VERSION/<network_id>/advertisers/<advertiser_id_from_network>/advertiser_campaigns/<advertiser_campaign_id_from_network>/promo_numbers_by_id/<promo_number_id_from_network>.json``
```
I noticed that in the Advertisers docs we fill in the advertiser_id in the example endpoint instead of subbing it out with <>
Also, on the tab we call it <id_from_network> but in the endpoint it is called <promo_number_id_from_network>. Might make sense to match these or fill in the example id_from_network
(also applies to delete and get)

collin-mcgrath replied:
Good point on both fronts.

collin-mcgrath committed:
WEB-4108 niapi promo numbers does not support createupdate by idfromnetwork: Fix syntax highlighting

collin-mcgrath committed:
WEB-4108 niapi promo numbers does not support createupdate by idfromnetwork: Use id_from_network per updated requests

collin-mcgrath commented:
@rjkearns Your comment is valid. I referenced the same template in bot of the `GET` panels. Fixed.

collin-mcgrath committed:
WEB-4108 niapi promo numbers does not support createupdate by idfromnetwork: Reference the correct template

collin-mcgrath committed:
WEB-4108 niapi promo numbers does not support createupdate by idfromnetwork: Use stubbed IDs in endpoint examples

collin-mcgrath commented:
@rjkearns @joroeinvoca CR addressed.

rjkearns reviewed (approved):
👍
